tests/generators/test_lollipop.py

```diff
@@ -0,0 +1,34 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestLollipopGraph(unittest.TestCase):
+    def test_lollipop_graph(self):
+        graph = retworkx.generators.lollipop_graph(17, 3)
+        self.assertEqual(len(graph), 20)
+        self.assertEqual(len(graph.edges()), 139)
+
+    def test_lollipop_graph_weights(self):
+        graph = retworkx.generators.lollipop_graph(
+            mesh_weights=list(range(17)), path_weights=list(range(17, 20))
+        )
+        self.assertEqual(len(graph), 20)
+        self.assertEqual(list(range(20)), graph.nodes())
+        self.assertEqual(len(graph.edges()), 139)
```

Comment:
For these tests it would be good to assert the edge lists are consistent with what's expected for a lolipop graph. Right now these tests are just asserting the node and edge counts match up but the graph could be completely wrong. Obviously if you do this it would probably be good to reduce the size (nobody is gonna want to manually work with a 139 element edge list :)

Reply:
Any suggestions how to do that? I was looking at the tests for the other graphs, but could not really figure out how to do that. From what I see it seems to me that the tests for the undirected graphs only check that the node and edge counts match up.

Reply:
The other tests probably need to be updated (or predate the `*edge_list()` methods which might have been added after some of the generators) that's something we can fix in a new PR. The easiest way would be to do something like:
```python
graph = retworkx.generators.lolipop_graph(3, 3)
edge_list = graph.edge_list()
expected_edge_list = [
    (0, 1),
    (0, 2),
    (1, 2),
    (2, 3),
    (3, 4),
    (4, 5)
]
self.assertEqual(edge_list, expected_edge_list)
```
or use `weighted_edge_list()` if you want to validate the weights are set correctly too